src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1662,10 +1753,532 @@ private static Entity tryReturnSource(CommandSource source, CommandArgs args) th
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String integerString = args.next();
+            try {
+                return new BigInteger(integerString);
+            } catch (NumberFormatException ex) {
+                throw args.createError(Text.of("Expected an integer, but input "+integerString+" was not"));
+            }
+        }
+    }
+
+    /**
+     * Expect an argument to be a valid {@link DataContainer}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement dataContainer(Text key) {
+        return new DataElement(key);
+    }
+
+    private static class DataElement extends RemainingJoinedStringsCommandElement {
+
+        protected DataElement(Text key) {
+            super(key, true);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String argument = (String) super.parseValue(source, args);
+            Callable<BufferedReader> reader = () -> new BufferedReader(new StringReader(argument));
+            ConfigurationLoader<? extends ConfigurationNode> loader = HoconConfigurationLoader.builder()
+                    .setSource(reader).build();
+            ConfigurationNode node;
+            try {
+                node = loader.load();
+            } catch (IOException ex) {
+                throw args.createError(Text.of("Node parsing failed: ", ex.getMessage()));
+            }
+            return DataTranslators.CONFIGURATION_NODE.translate(node);
+        }
+
+    }
+
+    /**
+     * Expect an argument to be a {@link UUID}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement uuid(Text key) {
+        return new UUIDElement(key);
+    }
+
+    private static class UUIDElement extends KeyElement {
+
+        protected UUIDElement(Text key) {
+            super(key);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            try {
+                return UUID.fromString(args.next());
+            } catch (IllegalArgumentException ex) {
+                throw args.createError(Text.of("Invalid UUID!"));
+            }
+        }
+
+    }
+
+    /**
+     * Expect an argument to be valid {@link Text}. Can use either JSON or color codes.
+     *
+     * @param key The key to store under
+     * @param complex If true, parsed as JSON text; if false, parsed as
+     *                color-coded text
+     * @param allRemaining If true, consumes all remaining arguments; if false,
+     *                     uses a single argument
+     * @return the argument
+     */
+    public static CommandElement text(Text key, boolean complex, boolean allRemaining) {
```

Comment:
Instead of a boolean for the formatting, I think you should just pass in a `TextSerializer` to give maximum flexibility.